File: Dummy/hawkshaw/library/datalayer/models/BatchThumbUploadRequest.java
package com.hawkshaw.library.datalayer.models;

import Y1.K;

public final class BatchThumbUploadRequest {
    private final Thumbnail thumb;

    public static final class Thumbnail {
        private final byte[] content;
        private final String name;
        private final String path;

        public Thumbnail(String str, String str2, byte[] bArr) {
            K.n(str, "name");
            K.n(str2, "path");
            K.n(bArr, "content");
            this.name = str;
            this.path = str2;
            this.content = bArr;
        }

        public final byte[] getContent() {
            return this.content;
        }

        public final String getName() {
            return this.name;
        }

        public final String getPath() {
            return this.path;
        }
    }

    public BatchThumbUploadRequest(Thumbnail thumbnail) {
        K.n(thumbnail, "thumb");
        this.thumb = thumbnail;
    }

    public final Thumbnail getThumb() {
        return this.thumb;
    }
}

File: app/src/main/java/com/hawkshaw/library/datalayer/models/BatchThumbUploadRequest.kt
package com.hawkshaw.library.datalayer.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Request model for batch thumbnail upload operation
 */
@Serializable
data class BatchThumbUploadRequest(
    @SerialName("thumb")
    val thumb: Thumbnail
) {
    /**
     * Thumbnail data structure
     */
    @Serializable
    data class Thumbnail(
        @SerialName("name")
        val name: String,
        
        @SerialName("path")
        val path: String,
        
        @SerialName("content")
        val content: ByteArray
    ) {
        override fun equals(other: Any?): Boolean {
            if (this === other) return true
            if (javaClass != other?.javaClass) return false

            other as Thumbnail

            if (name != other.name) return false
            if (path != other.path) return false
            if (!content.contentEquals(other.content)) return false

            return true
        }

        override fun hashCode(): Int {
            var result = name.hashCode()
            result = 31 * result + path.hashCode()
            result = 31 * result + content.contentHashCode()
            return result
        }
    }
} 